pybamm/models/submodels/thermal/current_collector/current_collector_full_thermal.py

```diff
@@ -0,0 +1,138 @@
+#
+# Class for N+1D thermal submodel which accounts for current collectors
+#
+import pybamm
+
+from .base_current_collector_thermal import BaseModel
+
+
+class BaseNplus1D(BaseModel):
+    """Base class for N+1D thermal submodel which accounts for current collectors.
+
+    Parameters
+    ----------
+    param : parameter class
+        The parameters to use for this submodel
+
+
+    **Extends:** :class:`pybamm.thermal.current_collector.BaseModel`
+    """
+
+    def __init__(self, param):
+        super().__init__(param)
+
+    def get_fundamental_variables(self):
+
+        T_av = pybamm.standard_variables.T_av
+
+        variables = self._get_standard_fundamental_variables(T_av)
+        return variables
+
+    def get_coupled_variables(self, variables):
+        variables.update(self._get_standard_coupled_variables(variables))
+        return variables
+
+    def _unpack(self, variables):
+        T_av = variables["X-averaged cell temperature"]
+        q = variables["Heat flux"]
+        Q_av = variables["X-averaged total heating"]
+        return T_av, q, Q_av
+
+
+class Full1plus1D(BaseNplus1D):
+    """Class for 1+1D thermal submodel"""
+
+    def __init__(self, param):
+        super().__init__(param)
+
+    def set_rhs(self, variables):
+        T_av, _, Q_av = self._unpack(variables)
+        self.rhs = {
+            T_av: (
```

Comment:
I thought full thermal would require the boundary temperature from the "whole_cell" domain?

Reply:
"full" here refers to T just being a function of y and z, so the BCs "left" and "right" correspond to the negative and positive tabs (here `T_av` lives on the current collector domain)

Reply:
Ah that makes sense.